sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_client_async.py

```diff
@@ -329,64 +420,138 @@ def from_connection_string(
+         not (>). True for inclusive and False for exclusive. This can be a dict with partition ID as the key and
+         bool as the value indicating whether the starting_position for a specific partition is inclusive or not.
+         This can also be a single bool value for all starting_position. The default value is False.
+        :paramtype starting_position_inclusive: bool or dict[str,bool]
+        :keyword on_error: The callback function that will be called when an error is raised during receiving
+         after retry attempts are exhausted, or during the process of load-balancing.
+         The callback takes two parameters: `partition_context` which contains partition information
+         and `error` being the exception. `partition_context` could be None if the error is raised during
+         the process of load-balance. The callback should be defined like: `on_error(partition_context, error)`.
+         The `on_error` callback will also be called if an unhandled exception is raised during
+         the `on_event` callback.
+        :paramtype on_error: Callable[[~azure.eventhub.aio.PartitionContext, Exception]]
+        :keyword on_partition_initialize: The callback function that will be called after a consumer for a certain
+         partition finishes initialization. It would also be called when a new internal partition consumer is created
+         to take over the receiving process for a failed and closed internal partition consumer.
+         The callback takes a single parameter: `partition_context`
+         which contains the partition information. The callback should be defined
+         like: `on_partition_initialize(partition_context)`.
+        :paramtype on_partition_initialize: Callable[[~azure.eventhub.aio.PartitionContext]]
+        :keyword on_partition_close: The callback function that will be called after a consumer for a certain
+         partition is closed. It would be also called when error is raised during receiving after retry attempts are
+         exhausted. The callback takes two parameters: `partition_context` which contains partition
+         information and `reason` for the close. The callback should be defined like:
+         `on_partition_close(partition_context, reason)`.
+         Please refer to :class:`CloseReason<azure.eventhub.CloseReason>` for the various closing reasons.
+        :paramtype on_partition_close: Callable[[~azure.eventhub.aio.PartitionContext, ~azure.eventhub.CloseReason]]
+        :rtype: None
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/async_samples/sample_code_eventhub_async.py
+                :start-after: [START eventhub_consumer_client_receive_batch_async]
+                :end-before: [END eventhub_consumer_client_receive_batch_async]
+                :language: python
+                :dedent: 4
+                :caption: Receive events from the EventHub.
+        """
+        assert max_batch_size is None or max_batch_size > 0
```

Comment:
Rather than an assertion error, we should probably raised a value error with a message.

Reply:
Changed to raise ValueError